pkg/controller/perconaservermongodb/status.go

```diff
@@ -233,3 +239,39 @@ func (r *ReconcilePerconaServerMongoDB) rsStatus(rsSpec *api.ReplsetSpec, cluste
 
 	return status, nil
 }
+
+func (r *ReconcilePerconaServerMongoDB) connectionEndpoint(cr *api.PerconaServerMongoDB) (string, error) {
+	if cr.Spec.Sharding.Enabled {
+		if mongos := cr.Spec.Sharding.Mongos; mongos.Expose.Enabled &&
+			mongos.Expose.ExposeType == corev1.ServiceTypeLoadBalancer {
+			return loadBalancerServiceEndpoint(r.client, cr.Name+"-mongos", cr.Namespace)
+		}
+		return cr.Name + "-mongos." + cr.Namespace + "." + cr.Spec.ClusterServiceDNSSuffix, nil
+	}
+
+	if rs := cr.Spec.Replsets[0]; rs.Expose.Enabled &&
+		rs.Expose.ExposeType == corev1.ServiceTypeLoadBalancer {
+		return loadBalancerServiceEndpoint(r.client, cr.Name+"-"+cr.Spec.Replsets[0].Name, cr.Namespace)
```

Comment:
in such case, I think we should return a comma-separated list of all load balancers.

like this
```
     addrs, err := psmdb.GetReplsetAddrs(c, cluster, rs, pods.Items)
     if err != nil {
         return nil, errors.Wrap(err, "get mongo addr")
     }
     return strings.Join(addrs, ","), nil
```

Reply:
ca6fa9c